specs/CustomDefaultDownloadLocation.md

```diff
@@ -0,0 +1,80 @@
+Custom Default Download Location
+===
+
+# Background
+This API allows you to set a custom default download location per profile. The
+user can still change the path through the Save As dialog.
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+# Examples
+```cpp
+HRESULT AppWindow::OnCreateCoreWebView2ControllerCompleted(
+    HRESULT result, ICoreWebView2Controller* controller)
+{
+    if (result == S_OK)
+    {
+        if (!m_webviewOption.downloadPath.empty())
+        {
+            Microsoft::WRL::ComPtr<ICoreWebView2Profile2> profile2;
+            CHECK_FAILURE(m_profile->QueryInterface(IID_PPV_ARGS(
+                &profile2)));
+            CHECK_FAILURE(profile2->put_DefaultDownloadPath(
+                m_webviewOption.downloadPath.c_str()));
+        }
+    }
+}
+```
+```c#
+void SetDefaultDownloadPathCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    var dialog = new TextInputDialog(
+        title: "Set Default Download Path",
+        description: "Enter the new default download path.",
+        defaultInput: WebViewProfile.DefaultDownloadPath);
+    if (dialog.ShowDialog() == true)
+    {
+        WebViewProfile.DefaultDownloadPath = dialog.Input.Text;
+    }
+}
+```
+# API Details
+```c#
+/// This is a continuation of the `ICoreWebView2Profile` interface.
+[uuid(DAF8B1F9-276D-410C-B481-58CBADF85C9C), object, pointer_default(unique)]
+interface ICoreWebView2Profile2 : ICoreWebView2Profile {
+  /// Gets the `DefaultDownloadPath` property. The default value is the
+  /// system default download path for the user.
+  [propget] HRESULT DefaultDownloadPath([out, retval] LPWSTR* value);
+
+  /// Sets the `DefaultDownloadPath` property. The value should be
+  /// an absolute path to a folder that the user and application can write to.
+  /// Returns `E_INVALIDARG` if the value is invalid, and the default download
+  /// path is not changed. If the directory does not yet exist, it is created
+  /// at the time of the next download. The user can still change the path
```

Comment:
If the specified download directory is a valid path but does not yet exist, and then at the time of the next download it cannot be created because the host app doesn't have permission to create the download directory, what happens? Does the download fail or fallback to some other path or something?